### PR TITLE
doc: CAN: Add CAN related changes to the 2.1 release notes

### PR DESCRIPTION
**Pull request by alexanderwachter**
Add notes about CAN related changes for this release.

### CODE REVIEW

**Review by dleach02 — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by karstenkoenig — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst [hunk 1]
```diff
@@ -26,12 +26,21 @@ Architectures
 Boards & SoC Support
 ********************
 
-* TBD
+* CAN support:
```
Comment (ioannisg) [inline doc/releases/release-notes-2.1.rst]: why is this directly under Boards & SoC Support?

Comment (alexanderwachter) [reply doc/releases/release-notes-2.1.rst]: Because its CAN support for a board.
Tho other one is a shield.

**Review by dbkinder — APPROVED** [review]

**Review by nixward — APPROVED** [review]

**Comment by ioannisg** [comment]
Needs rebase @alexanderwachter

File: doc/releases/release-notes-2.1.rst [hunk 2]
```diff
@@ -90,6 +92,12 @@ Drivers and Sensors
   * Added Atmel AT24 (and compatible) I2C EEPROM driver
   * Added Atmel AT25 (and compatible) SPI EEPROM driver
   * Added native_posix EEPROM emulation driver
+* CAN
```
Comment (ioannisg) [inline doc/releases/release-notes-2.1.rst]: maybe a new line here @alexanderwachter ?

Comment (alexanderwachter) [reply doc/releases/release-notes-2.1.rst]: Added.

File: doc/releases/release-notes-2.1.rst [hunk 3]
```diff
@@ -91,6 +93,13 @@ Drivers and Sensors
   * Added Atmel AT25 (and compatible) SPI EEPROM driver
   * Added native_posix EEPROM emulation driver
 
+* CAN
```
Comment (ioannisg) [inline doc/releases/release-notes-2.1.rst]: Ah, I know i am too picky here, but is it possible to put CAN section before EEPROM?, so it is in alphabetical order

Comment (alexanderwachter) [reply doc/releases/release-notes-2.1.rst]: No problem!

**Review by ioannisg — APPROVED** [review]

**Pull request merged**